### ISSUE TITLE
Use Python's built-in initgroups always

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5861]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5861|
|Type|enhancement|
|Created|2012-08-13 14:51:52Z|
|Branch|https://github.com/twisted/twisted/tree/less-c-initgroups-5861|

Python 2.7 and 3.3 include `os.initgroups`, which we can use instead of building our own C extension.

<details><summary>Searchable metadata</summary>

```
trac-id__5861 5861
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__EndOfLifePython2_6 EndOfLifePython2.6
branch__branches_less_c_initgroups_5861 branches/less-c-initgroups-5861
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1344869512000000 1344869512000000
changetime__1441853898291004 1441853898291004
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

Adding to minimal python 3.3 milestone since used by process.

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_2"></a>|@vperic commented|
|-|-|

Duplicate of [#4920](https://github.com/twisted/twisted/issues/4920)? (and yes, I know that ticket is assigned to me - I got stuck when exarkun asked me to write tests and I didn't know how)

BTW, what is this new "Python 3.3 Minimal" milestone? Seems to be relevant to my project (such as it is), but I haven't managed to find any discussion anywhere about it.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst set status to `closed`|
|-|-|

Duplicate of [#4920](https://github.com/twisted/twisted/issues/4920), yes. See http://twistedmatrix.com/trac/wiki/Plan/Python3 for details.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

This is a 2.6 thing.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `reopened`|
|-|-|

Bah this actually wasn't open. But rather than file another duplicate, might as well use this one!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl commented|
|-|-|

(In [45634]) Branching to less-c-initgroups-5861.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl commented|
|-|-|

Okay! Removed the old ```_initgroups.c```, and simplified the tests considerably.

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban set owner to @hawkowl<br>@adiroiban set status to `new`|
|-|-|

Please add a bit more details to the scope and changes to this branch.

So with this change we drop support for python 2.6 ? 

Why was the _initgroups.c remove.

From what I can see the code from trunk was already using the builting initgroups when available.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_9"></a>|@hawkowl removed owner|
|-|-|

Hi Adi,

We've dropped support for 2.6 already officially (see last release announcement), this just cleans up some things that are not required.

```_initgroups.c``` is removed because it was only required for 2.6 support -- it is merged into stdlib now. I've altered the ticket slightly.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_10"></a>|@adiroiban set owner to @hawkowl|
|-|-|

There is one error which I think can be fixed 

```
************* Module twisted.python.util
W9204:644,4:initgroups: Missing epytext markup @return for return value
```

please fix and merge.

thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_11"></a>|@hawkowl set status to `closed`|
|-|-|

(In [45662]) Merge less-c-initgroups-5861: Use Python's built-in initgroups always

Author: hawkowl
Reviewer: adiroiban
Fixes: [#5861](https://github.com/twisted/twisted/issues/5861)
